pkg/controller/backup-operator/s3_backup.go

```diff
@@ -23,20 +24,35 @@ import (
 	"github.com/coreos/etcd-operator/pkg/backup/backupapi"
 	"github.com/coreos/etcd-operator/pkg/backup/writer"
 	"github.com/coreos/etcd-operator/pkg/util/awsutil/s3factory"
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
 
 	"k8s.io/client-go/kubernetes"
 )
 
 // TODO: replace this with generic backend interface for other options (PV, Azure)
 // handleS3 backups up etcd cluster to s3 and return s3 path for the backup file.
-func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, namespace, clusterName string) (string, error) {
+func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, operatorSecret, namespace, clusterName string) (string, error) {
 	cli, err := s3factory.NewClientFromSecret(kubecli, namespace, s3.AWSSecret)
 	if err != nil {
 		return "", err
 	}
 	defer cli.Close()
+
+	var tlsConfig *tls.Config
+	if len(operatorSecret) != 0 {
+		d, err := k8sutil.GetTLSDataFromSecret(kubecli, namespace, operatorSecret)
+		if err != nil {
+			return "", err
```

Comment:
Probably wrap the error:
`fmt.Errorf("failed to get TLS data from secret(%v): %v", operatorSecret, err)`